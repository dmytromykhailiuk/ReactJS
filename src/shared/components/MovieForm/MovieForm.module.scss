@import "scss/mixins";

.buttons {
  display: block;
  margin-top: 50px;
    
  & > * {
    display: block;
    
    &:first-child {
      margin-right: 0px;
      margin-bottom: 20px;
    }
  }

  @include medium-screen {
    margin-top: 80px;
    display: flex;
    justify-content: flex-end;

    & > * {
      &:first-child {
        margin-right: 20px;
      }
    }
  }
}