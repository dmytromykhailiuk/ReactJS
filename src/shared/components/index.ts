export * from "./Footer";
export * from "./Logo";
export * from "./Button";
export * from "./Input";
export * from "./ThreeDotsIcon";
export * from "./ErrorBoundary";
