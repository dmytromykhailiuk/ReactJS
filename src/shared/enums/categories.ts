export enum Categories {
  ALL = "ALL",
}
