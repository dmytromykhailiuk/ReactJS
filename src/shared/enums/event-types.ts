export enum EventTypes {
  CLICK = "click",
  SCROLL = "scroll",
}
