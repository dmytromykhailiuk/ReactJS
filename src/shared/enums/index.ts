export * from "./button-types";
export * from "./button-size";
export * from "./categories";
export * from "./sorting-options";
export * from "./movie-form-values";
export * from "./router-paths";
export * from "./modal-types";
