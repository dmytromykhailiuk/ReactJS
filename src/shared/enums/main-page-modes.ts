export enum MainPageModes {
  OVERVIEW = "overview",
  EDIT = "edit",
  CREATE = "create",
  DELETE = "delete",
}
