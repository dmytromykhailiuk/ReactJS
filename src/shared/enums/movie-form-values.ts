export enum MovieFormValues {
  MOVIE_ID = "id",
  TITLE = "title",
  RELEASE_DATE = "releaseDate",
  MOVIE_URL = "url",
  CATEGORY = "category",
  OVERVIEW = "overview",
  RUNTIME = "runtime",
  DURATION = "duration",
  RATING = "rating",
}
