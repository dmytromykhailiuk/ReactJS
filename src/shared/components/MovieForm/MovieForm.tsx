import React from "react";
import classes from "./MovieForm.module.scss";
import { Formik } from 'formik';
import { Movie } from "models/";
import { Input, Dropdown, Button } from "../";
import { categories } from "mocks";
import { ButtonTypes, Categories } from "shared/enums";
import { generateId } from "shared/helpers";
import * as Yup from 'yup';

interface MovieFormProps {
  onSubmitForm: (movie: Movie) => void;
  movie?: Movie;
  submitButtonLabel?: string;
}

interface  FormData {
  title: string;
  releaseDate: string;
  movieUrl: string;
  genre: Categories[];
  overview: string;
  runtime: string;
}

enum MovieFormValues {
  MOVIE_ID = 'movieId',
  TITLE = 'title',
  RELEASE_DATE = 'releaseDate',
  MOVIE_URL = 'movieUrl',
  GENRE = 'genre',
  OVERVIEW = 'overview',
  RUNTIME = 'runtime'
}

const URL_REGEX = /^((https?|ftp):\/\/)?(www.)?(((([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(%[\da-f]{2})|[!\$&'\(\)\*\+,;=]|:)*@)?(((\d|[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\.(\d|[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\.(\d|[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\.(\d|[1-9]\d|1\d\d|2[0-4]\d|25[0-5]))|((([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])*([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])))\.)+(([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])*([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])))\.?)(:\d*)?)(\/((([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(%[\da-f]{2})|[!\$&'\(\)\*\+,;=]|:|@)+(\/(([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(%[\da-f]{2})|[!\$&'\(\)\*\+,;=]|:|@)*)*)?)?(\?((([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(%[\da-f]{2})|[!\$&'\(\)\*\+,;=]|:|@)|[\uE000-\uF8FF]|\/|\?)*)?(\#((([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(%[\da-f]{2})|[!\$&'\(\)\*\+,;=]|:|@)|\/|\?)*)?$/i;

const ValidationSchema = Yup.object().shape({
  [MovieFormValues.TITLE]: Yup.string().required('Title is required'),
  [MovieFormValues.RELEASE_DATE]: Yup.string().required('Relese date is required'),
  [MovieFormValues.MOVIE_URL]: Yup.string().matches(URL_REGEX, 'Enter a valid url').required('Movie URL is required'),
  [MovieFormValues.GENRE]: Yup.array().min(1, "Select at least one genre to proceed").required("Ganre is required"),
  [MovieFormValues.OVERVIEW]: Yup.string().required('Owerview is required'),
  [MovieFormValues.RUNTIME]: Yup.string().required('Runtime is required')
});

const MovieForm: React.FC<MovieFormProps> = ({ onSubmitForm, movie = {}, submitButtonLabel = "SAVE" }) => {
  return (
    <Formik
      initialValues={{
        [MovieFormValues.TITLE]: movie.title || '',
        [MovieFormValues.RELEASE_DATE]: movie.releaseDate || '',
        [MovieFormValues.MOVIE_URL]: movie.movieUrl || '',
        [MovieFormValues.GENRE]: movie.genre || [],
        [MovieFormValues.OVERVIEW]: movie.overview || '',
        [MovieFormValues.RUNTIME]: movie.runtime || '',
      }}
      // validateOnMount={true}
      validationSchema={ValidationSchema}
      validateOnBlur={true}
      onSubmit={(values: FormData) => {
        onSubmitForm({...values, movieId: movie.movieId || generateId()});
      }}
    >
      {(formik) => (
        <form onSubmit={formik.handleSubmit}>
          {movie.movieId && <Input 
            disabled={true}
            name={MovieFormValues.MOVIE_ID}
            type="text"
            label="MOVIE ID"
            value={movie.movieId}
          />}
          <Input 
            name={MovieFormValues.TITLE}
            type="text"
            onChange={formik.handleChange}
            onBlur={formik.handleBlur}
            placeholder='Title here'
            label="TITLE"
            error={formik.errors.title && formik.touched.title ? formik.errors[MovieFormValues.TITLE] : ''}
            value={formik.values[MovieFormValues.TITLE]}
          />
          <Input 
            name={MovieFormValues.RELEASE_DATE}
            type="date"
            onChange={formik.handleChange}
            onBlur={formik.handleBlur}
            placeholder='Select Date'
            label="RELEASE DATE"
            error={formik.errors[MovieFormValues.RELEASE_DATE] && formik.touched[MovieFormValues.RELEASE_DATE] ? formik.errors[MovieFormValues.RELEASE_DATE] : ''}
            value={formik.values[MovieFormValues.RELEASE_DATE]}
          />
          <Input 
            name={MovieFormValues.MOVIE_URL}
            type="text"
            onChange={formik.handleChange}
            onBlur={formik.handleBlur}
            placeholder='Movie URL here'
            label="MOVIE URL"
            error={formik.errors[MovieFormValues.MOVIE_URL] && formik.touched[MovieFormValues.MOVIE_URL] ? formik.errors[MovieFormValues.MOVIE_URL] : ''}
            value={formik.values[MovieFormValues.MOVIE_URL]}
          />

          <Dropdown 
            label="GENRE" 
            placeholder="Select Genre" 
            name={MovieFormValues.GENRE} 
            options={categories}
            error={formik.errors[MovieFormValues.GENRE] && formik.touched[MovieFormValues.GENRE] ? formik.errors[MovieFormValues.GENRE] : ''}
          />

          <Input 
            name={MovieFormValues.OVERVIEW}
            type="text"
            onChange={formik.handleChange}
            onBlur={formik.handleBlur}
            placeholder='Overview here'
            label="OVERVIEW"
            error={formik.errors[MovieFormValues.OVERVIEW] && formik.touched[MovieFormValues.OVERVIEW] ? formik.errors[MovieFormValues.OVERVIEW] : ''}
            value={formik.values[MovieFormValues.OVERVIEW]}
          />
          <Input 
            name={MovieFormValues.RUNTIME}
            type="text"
            onChange={formik.handleChange}
            onBlur={formik.handleBlur}
            placeholder='Runtime here'
            label="RUNTIME"
            error={ formik.errors[MovieFormValues.RUNTIME] && formik.touched[MovieFormValues.RUNTIME] ? formik.errors[MovieFormValues.RUNTIME] : ''}
            value={formik.values[MovieFormValues.RUNTIME]}
          />
          <div className={classes.buttons}>
            <Button 
              type={ButtonTypes.SECONDARY} 
              onButtonClicked={() => formik.resetForm()} 
            >
              RESET
            </Button>
            <Button
              isSubmit={true}
              type={ButtonTypes.PRIMARY}
              disabled={!formik.isValid || !formik.dirty}
            >
              { submitButtonLabel }
            </Button>
          </div>
        </form>
      )}
     </Formik>
  )
}

export default MovieForm;
