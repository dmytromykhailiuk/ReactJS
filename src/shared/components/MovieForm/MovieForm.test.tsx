import { mount, shallow } from "enzyme";
import React from "react";
import toJson from 'enzyme-to-json';
import MovieForm from "./MovieForm";
import { movies } from "mocks/movies.mock";
import { Formik } from "formik";
import { MovieFormValues } from "shared/enums";

function delay(): Promise<void> {
  return new Promise((r) => setTimeout(r, 1000));
}

describe("MovieForm", () => {
  it("should match snepshot", () => {
    const wrapper = shallow(
      <MovieForm
        onSubmitForm={() => {}}
        movie={movies[0]}
      />
    )
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it("should call 'onSubmitForm' when submit", () => {
    const onSubmitForm = jest.fn();
    const wrapper = shallow(
      <MovieForm
        onSubmitForm={onSubmitForm}
        movie={movies[0]}
      />
    )
    wrapper.find(Formik).simulate('submit');
    expect(onSubmitForm).toHaveBeenCalled();
  })

  it("should reset form", () => {
    const wrapper = mount(
      <MovieForm
        onSubmitForm={() => {}}
      />
    )
    const input = wrapper.find("input[name='title']");

    input.simulate('change', {
      persist: () => {},
      target: {
        name: 'title',
        value: 'title'
      }
    });

    wrapper.find("button").first().simulate("click")
     
    expect((input.getDOMNode() as any).value).toEqual('');
  })

  it("should show input with id when movie has id", () => {
    const wrapper = mount(
      <MovieForm
        onSubmitForm={() => {}}
        movie={movies[0]}
      />
    )
     
    expect(wrapper.exists("input[name='id']")).toBeTruthy();
  })

  it("should show error in MovieFormValues.TITLE field", async () => {
    const wrapper = mount(
      <MovieForm
        onSubmitForm={() => {}}
      />
    )

    const inputTitle = wrapper.find(`input[name='${MovieFormValues.TITLE}']`);

    inputTitle.simulate('focus').simulate('blur', {
      persist: () => {},
      target: {
        name: MovieFormValues.TITLE,
        value: ''
      }
    });

    await delay();

    expect((inputTitle.getDOMNode().nextSibling as HTMLElement).textContent).toBeTruthy();
  })

  it("should show error in MovieFormValues.RELEASE_DATE field", async () => {
    const wrapper = mount(
      <MovieForm
        onSubmitForm={() => {}}
      />
    )

    const inputTitle = wrapper.find(`input[name='${MovieFormValues.RELEASE_DATE}']`);

    inputTitle.simulate('focus').simulate('blur', {
      persist: () => {},
      target: {
        name: MovieFormValues.RELEASE_DATE,
        value: ''
      }
    });

    await delay();

    expect((inputTitle.getDOMNode().nextSibling as HTMLElement).textContent).toBeTruthy();
  })

  it("should show error in MovieFormValues.MOVIE_URL field", async () => {
    const wrapper = mount(
      <MovieForm
        onSubmitForm={() => {}}
      />
    )

    const inputTitle = wrapper.find(`input[name='${MovieFormValues.MOVIE_URL}']`);

    inputTitle.simulate('focus').simulate('blur', {
      persist: () => {},
      target: {
        name: MovieFormValues.MOVIE_URL,
        value: ''
      }
    });

    await delay();

    expect((inputTitle.getDOMNode().nextSibling as HTMLElement).textContent).toBeTruthy();
  })

  it("should show error in MovieFormValues.OVERVIEW field", async () => {
    const wrapper = mount(
      <MovieForm
        onSubmitForm={() => {}}
      />
    )

    const inputTitle = wrapper.find(`input[name='${MovieFormValues.OVERVIEW}']`);

    inputTitle.simulate('focus').simulate('blur', {
      persist: () => {},
      target: {
        name: MovieFormValues.OVERVIEW,
        value: ''
      }
    });

    await delay();

    expect((inputTitle.getDOMNode().nextSibling as HTMLElement).textContent).toBeTruthy();
  })

  it("should show error in MovieFormValues.RATING field", async () => {
    const wrapper = mount(
      <MovieForm
        onSubmitForm={() => {}}
      />
    )

    const inputTitle = wrapper.find(`input[name='${MovieFormValues.RATING}']`);

    inputTitle.simulate('focus').simulate('change', {
      persist: () => {},
      target: {
        name: MovieFormValues.RATING,
        value: -20
      }
    }).simulate('blur', {
      persist: () => {},
      target: {
        name: MovieFormValues.RATING,
        value: -20
      }
    });

    await delay();

    expect((inputTitle.getDOMNode().nextSibling as HTMLElement).textContent).toBeTruthy();
  })

  it("should show error in MovieFormValues.RUNTIME field", async () => {
    const wrapper = mount(
      <MovieForm
        onSubmitForm={() => {}}
      />
    )

    const inputTitle = wrapper.find(`input[name='${MovieFormValues.RUNTIME}']`);

    inputTitle.simulate('focus').simulate('blur', {
      persist: () => {},
      target: {
        name: MovieFormValues.RUNTIME,
        value: -20
      }
    });

    await delay();

    expect((inputTitle.getDOMNode().nextSibling as HTMLElement).textContent).toBeTruthy();
  })
})
