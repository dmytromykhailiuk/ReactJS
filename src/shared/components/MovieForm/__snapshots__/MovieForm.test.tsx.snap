// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieForm should match snepshot 1`] = `
<Formik
  initialValues={
    Object {
      "genres": Array [
        "Drama",
        "Romance",
      ],
      "overview": "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
      "poster_path": "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
      "release_date": "2018-02-07",
      "runtime": 106,
      "title": "Fifty Shades Freed",
      "vote_average": 6.1,
    }
  }
  onSubmit={[Function]}
  validateOnBlur={true}
  validationSchema={
    ObjectSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_blacklistError": undefined,
      "_excludedEdges": Array [],
      "_nodes": Array [
        "vote_average",
        "runtime",
        "overview",
        "genres",
        "poster_path",
        "release_date",
        "title",
      ],
      "_sortErrors": [Function],
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_whitelistError": undefined,
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {},
      "fields": Object {
        "genres": ArraySchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_mutate": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "min": true,
            "required": true,
          },
          "innerType": undefined,
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "array",
        },
        "overview": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_mutate": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "poster_path": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_mutate": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "matches": false,
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "release_date": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_mutate": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "runtime": NumberSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_mutate": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "min": true,
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "number",
        },
        "title": StringSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_mutate": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "required": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "required",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "string",
        },
        "vote_average": NumberSchema {
          "_blacklist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_blacklistError": undefined,
          "_typeError": [Function],
          "_whitelist": ReferenceSet {
            "list": Set {},
            "refs": Map {},
          },
          "_whitelistError": undefined,
          "conditions": Array [],
          "deps": Array [],
          "exclusiveTests": Object {
            "max": true,
            "min": true,
          },
          "spec": Object {
            "abortEarly": true,
            "nullable": false,
            "presence": "optional",
            "recursive": true,
            "strict": false,
            "strip": false,
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
          "type": "number",
        },
      },
      "spec": Object {
        "abortEarly": true,
        "nullable": false,
        "presence": "optional",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
      "type": "object",
    }
  }
>
  <Component />
</Formik>
`;
